protos/sources/vega/api/v1/core.proto

```diff
@@ -343,3 +348,61 @@ message LastBlockHeightResponse {
   // the network chain id from which the block comes from
   string chain_id = 8;
 }
+
+// A request to retrieve the spam statistics of a party for the given epoch
+message GetSpamStatisticsRequest {
+  string party_id = 1;
+}
+
+// The statistics for a given spam policy
+message SpamStatistic {
+  // The current transaction count received from the party during this epoch for this policy
+  string count_for_epoch = 1;
+  // The maximum number of transactions allowed for this policy in an epoch
+  string max_for_epoch = 2;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 4;
+}
+
+// The vote statistics for the voting spam policies
+// which are calculated as a ratio of the total votes
+// that have been rejected.
+message VoteSpamStatistic {
+  string proposal = 1;
+  // The current vote count received from the party for the given policy during this epoch
+  string count_for_epoch = 2;
+  // The maximum number of votes per proposal allowed in an epoch
+  string max_for_epoch = 3;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 4;
+}
+
+// The PoW state for a given block
+message PoWBlockState {
+  uint64 block_height = 1;
+  uint64 transactions_seen = 2;
+  uint64 difficulty = 3;
```

Comment:
I think it's worth adding here a few params for the block - the difficulty and the increaseDifficulty values in particular.

Reply:
block hash is a good idea. 
a couple of reasons - the wallet may not need to cache historical values of difficulty so if anyone wants to generate another transaction for a particular block having the difficulty param here is useful. In addition what I thought of originally is an optimisation where given the required difficulty and the observed one, the wallet can optimise the pow calculation to not increase the difficulty if not needed (if got lucky in earlier rounds and got increased difficulty).

Reply:
Core's calculation of what difficulty you need to use can take into account luckily getting high difficulty too. I just think if we ever change in the future the way the difficulty is calculated its a single core change, no big deal. If we let wallet's calculate then its a breaking-change that could affects many many people's code.

I don't really follow your point about having to cache historical values? Got an example?

Reply:
You've remembered it correctly (or atleast remembered it the same as me). 
What I'm saying is that a wallet (ours, or someone elses implementation) will read these values from the API and write some function `f(difficulty, increaseDifficulty, transactions_seen) -> difficulty_to_aim_for_now` 

So we might as well just have our API do `f()` for us and just return `difficulty_to_aim_for_now`

Reply:
we could, I wanted to keep it simple...

Reply:
Ok, fair enough lets keep it simple!